synapse/storage/databases/main/events.py

```diff
@@ -2249,11 +2175,6 @@ def _set_push_actions_for_event_and_users_txn(
     def _remove_push_actions_for_event_id_txn(
         self, txn: LoggingTransaction, room_id: str, event_id: str
     ) -> None:
-        # Sad that we have to blow away the cache for the whole room here
-        txn.call_after(
-            self.store.get_unread_event_push_actions_by_room_for_user.invalidate,
-            (room_id,),
-        )
```

Comment:
To check the review box that everything is accounted for, all of these removals are accounted for in `_invalidate_caches_for_event` and `_invalidate_state_caches` ✅

`_persist_events_txn` takes care of calling `_invalidate_caches_for_event` and `_update_current_state_txn` -> `_invalidate_state_caches_and_stream` -> `_invalidate_state_caches`

And `_invalidate_caches_for_event`/`_invalidate_state_caches` are already called over replication.